### ISSUE TITLE
Bug: Types not checked properly when variadic function calls non-variadic function

### ISSUE THREAD

**Issue by maxrothman** [python/mypy#6308]
* Mypy: 0.660
* Python: 3.7.2

Given the following code:
```
def variadic(*args: str, **kwargs: str) -> None:
    nonvariadic(*args, **kwargs)

def nonvariadic(a: str, b: str) -> None:
    print(a, b)

variadic()
```
Running `mypy --strict` yields no warnings. There's an obvious runtime error though:
```
TypeError: nonvariadic() missing 2 required positional arguments: 'a' and 'b'
```

Using a "proxy function" that takes `*args, **kwargs` is a pretty common pattern in real-world code, so this seems like something mypy should support.

**Comment by JelleZijlstra** [comment]
The general reason that mypy doesn't catch this is that type checking is local to individual functions, and there's no obvious type error in either function taken by itself.

The problem you're pointing out is real, but unfortunately there doesn't appear to be a simple solution. python/typing#270 has a concrete proposal that would cover your use case, so closing this issue in favor of the typing one.
